### ISSUE TITLE
 The training loss of the affine registration network keeps going up

### ISSUE THREAD

**Issue by XiaotianJia** [junyuchen245/TransMorph_Transformer_for_Medical_Image_Registration#7]
Thank you so much for sharing your outstanding work, which has helped me a lot. But when I use my own data to train the affine registration network, the training loss keeps going up. I have checked my training input and there is no problem. I set batch size to 1 and the learning rate to 0.00001. Do you know what the problem might be?

**Comment by junyuchen245** [comment]
Hi @XiaotianJia ,

I've tried the affine network again for 8 epochs and it seemed to be working fine:
<img src="https://user-images.githubusercontent.com/17306891/159197161-776b096c-0e7d-43ec-a8d6-c6de439dcf93.jpg" width=500>

There were a few typos in the script that I made when sorting the code; I've corrected them now and they should have no effect on performance or loss while training. One suggestion I would offer is to experiment with other loss functions, depending on the modality you are working on, but I've found that [MIND loss](https://github.com/junyuchen245/TransMorph_Transformer_for_Medical_Image_Registration/blob/9a681b054c3d6500211c2a0f804c936f86091482/TransMorph_affine/losses.py#L522) works the best.

Junyu

**Comment by ywzcode** [comment]
Hi, thank you for the great work. Somehow, I also have the same problem on the OASIS dataset using TransMorph default config. 
The loss is increasing and val DSC score is around 0.6. I'm using Pytorch 1.10.1 with cuda 11.

**Comment by ywzcode** [comment]
@junyuchen245 Thanks for your reply. It is the deformable model. I used the scripts and the OASIS data you provided. 

I just tried PyTorch 1.9, and it works now. I guess it is the PyTorch version problem. 

Thanks

**Comment by junyuchen245** [comment]
@ywzcode Sounds good! Thanks for letting me know.

Junyu
